nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -501,7 +517,7 @@ contract StakingEscrow is Issuer {
             policyManager.register(_staker, getCurrentPeriod());
         }
         info.value = info.value.add(_value);
-        token.safeTransferFrom(_staker, address(this), _value);
+        token.safeTransferFrom(_payer, address(this), _value);
```

Comment:
Why are you introducing a `_payer` here, apart from the `_staker`?

Reply:
I split staker and who transfer tokens (payer) inside internal method because worklock will call deposit in escrow and so it is payer, but staker will have other address as a worklock user. Also added method if somebody wants stake token for anyone else